### PR TITLE
refactor: handle token transfers

### PR DESCRIPTION
**Pull request by alexbarnsley**
## Summary

https://app.clickup.com/t/86dvatz0f

## Checklist

-   [ ] I checked my UI changes against the design and there are no notable differences
-   [ ] I checked my UI changes for any responsiveness issues
-   [ ] I checked my UI changes in light AND dark mode
-   [ ] I checked my (code) changes for obvious issues, debug statements and commented code
-   [ ] I opened a corresponding card on Clickup for any remaining TODOs in my code
-   [ ] I added a short description on how to test this PR _(if necessary)_
-   [ ] Documentation _(if necessary)_
-   [ ] Tests _(if necessary)_
-   [ ] Ready to be merged

### CODE REVIEW

File: resources/views/components/transaction/page/token.blade.php [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        @else
+            <x-transaction.page.section-detail.row
+                :title="trans('pages.transaction.header.value')"
+                :transaction="$transaction"
+            >
+                <livewire:fiat-value
+                    :amount="$transaction->amountWithFee()"
+                    :timestamp="$transaction->model()->timestamp"
+                />
+            </x-transaction.page.section-detail.row>
```
Comment (ItsANameToo) [inline resources/views/components/transaction/page/token.blade.php:38]: this currently shows `ARK` for the token, we'll have to consider that the token will have a different name

File: app/Services/ContractPayload.php [hunk 2]
```diff
@@ -0,0 +1,37 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Services;
+
+use ArkEcosystem\Crypto\Utils\Address;
+
+final class ContractPayload
```
Comment (ItsANameToo) [inline app/Services/ContractPayload.php:9]: candidate for php sdk?

File: resources/views/components/transaction/page/token.blade.php [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    >
+        <x-transaction.page.section-detail.address
+            :address="\App\Services\ContractPayload::decodeAddress($transaction->methodArguments()[0])"
+            class="inline-block"
+        />
+    </x-transaction.page.section-detail.row>
+
+    <x-transaction.page.section-detail.row
+        :title="trans('pages.transaction.header.amount')"
+        :value="ExplorerNumberFormatter::weiToArk(\App\Services\ContractPayload::decodeUnsignedInt($transaction->methodArguments()[1]))"
```
Comment (ItsANameToo) [inline resources/views/components/transaction/page/token.blade.php:16]: kinda ugly having all of this in the template 🤔 also hard to follow with the indexes. Maybe change them to an enum so it gives some context as to what that position includes?

**Review by ItsANameToo — CHANGES_REQUESTED** [review]

File: resources/views/components/transaction/page/token.blade.php [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    >
+        <x-transaction.amount :transaction="$transaction" />
+    </x-transaction.page.section-detail.row>
+
+    @if (Network::canBeExchanged())
+        @if (ExplorerNumberFormatter::isFiat(Settings::currency()) && ExchangeRate::convertNumerical($transaction->amountWithFee(), $transaction->model()->timestamp) < 0.01)
+            <x-transaction.page.section-detail.row
+                :title="trans('pages.transaction.header.value')"
+                :value="'<'.ExplorerNumberFormatter::currency(0.01, Settings::currency())"
```
Comment (ItsANameToo) [inline resources/views/components/transaction/page/token.blade.php:25]: this should be hidden if the token itself has no monetary value. This will be irregardless of whether the network itself has monetary value 🤔

File: resources/views/components/transaction/page/token.blade.php (hunk 1 above)
Comment (alexbarnsley) [reply resources/views/components/transaction/page/token.blade.php:38]: https://app.clickup.com/t/86dvbfz2x - added a card, as I still need to figure out how to pull that data from the contract

File: resources/views/components/transaction/page/token.blade.php (hunk 4 above)
Comment (alexbarnsley) [reply resources/views/components/transaction/page/token.blade.php:25]: I'll remove it completely for now and handle it later on. I'm not too sure how to determine if it has monetary value just yet :thinking: since we currently use jobs to cache the monetary value of ARK

Comment (alexbarnsley) [reply resources/views/components/transaction/page/token.blade.php]: https://app.clickup.com/t/86dvbg010

File: app/Services/ContractPayload.php (hunk 2 above)
Comment (alexbarnsley) [reply app/Services/ContractPayload.php:9]: Indeedy - I've added it to https://app.clickup.com/t/86dvb6env

**Review by ItsANameToo — APPROVED** [review]
